catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/forms/SearchFormsApplicationTest.java

```diff
@@ -0,0 +1,438 @@
+        .when(MOCK_FRAMEWORK)
+        .create(requestCaptor.capture());
+
+    // Execute
+    int statusCode =
+        RestAssured.given()
+            .header(CONTENT_IS_JSON)
+            .content(formRequestJson)
+            .post(localhostFormsUrl)
+            .statusCode();
+    assertThat(statusCode, is(200));
+    assertThat(getCapturedXml(), is(formFilterXml));
+  }
+
+  @Test
+  public void testXmlToJson()
+      throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+    // Prepare
+    QueryTemplateMetacard queryTemplateMetacard =
+        new QueryTemplateMetacard(CANNED_TITLE, CANNED_DESCRIPTION, CANNED_ID);
+    queryTemplateMetacard.setFormsFilter(formFilterXml);
+    queryTemplateMetacard.setCreatedDate(Date.from(Instant.parse(CANNED_ISO_DATE)));
+    queryTemplateMetacard.setModifiedDate(Date.from(Instant.parse(CANNED_ISO_DATE)));
+
+    QueryResponseImpl response =
+        new QueryResponseImpl(new QueryRequestImpl(new QueryImpl(Filter.INCLUDE)));
+    response.addResult(new ResultImpl(queryTemplateMetacard), true);
+
+    doReturn(response).when(MOCK_FRAMEWORK).query(any());
+
+    // Execute
+    String json =
+        RestAssured.given().header(CONTENT_IS_JSON).get(localhostFormsUrl).body().asString();
+    assertThat(json, is(formResponseJson));
+  }
+
+  private String getCapturedXml() {
+    Metacard searchForm = requestCaptor.getValue().getMetacards().get(0);
+    return ((QueryTemplateMetacard) searchForm).getFormsFilter();
+  }
+
+  /**
+   * Given a pretty JSON string, flatten out the string to enable string comparison operations
+   * regardless of formatting.
+   *
+   * @param json the input JSON string to flatten to a single line.
+   * @return a single line string of the given JSON.
+   */
+  private static String removePrettyPrintingOnJson(String json) {
+    return json.replaceAll("\\h|\\v", "");
+  }
+
+  /**
+   * Given a pretty XML string, flatten out the string to enable string comparison operations
+   * regardless of formatting.
+   *
+   * @param xml the input XML string to flatten to a single line.
+   * @return a single line string of the given XML.
+   */
+  private static String removePrettyPrintingOnXml(String xml) {
+    try {
+      TransformerFactory factory = TransformerFactory.newInstance();
+      Source xslt = new StreamSource(getResourceFile("inline-xml.xslt"));
+      Transformer transformer = factory.newTransformer(xslt);
+
+      Source text = new StreamSource(new ByteArrayInputStream(xml.getBytes()));
+      StringWriter destination = new StringWriter();
+
+      transformer.transform(text, new StreamResult(destination));
+
+      // Replace the header that the XSLT transform removed
+      return destination
+          .toString()
+          .replace(
+              "<?xml version=\"1.0\" encoding=\"UTF-8\"?>",
+              "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>");
```

Comment:
✏️ If anyone has any tips on how to bake this step into the actual transform I'm definitely willing to fix it. But based upon this SO post I do not think it's possible: 
https://stackoverflow.com/questions/29165895/preserving-the-xml-header-in-an-xslt-transform/29166106#29166106